youtube_dl/extractor/iheartradio.py

```diff
@@ -0,0 +1,213 @@
+            'accessToken': 'anon',
+            'accessTokenType': 'anon',
+            'deviceId': random_device_id,
+            'deviceName': "web-desktop",
+            'host': "webapp.WW",
+            'oauthUuid': random_oauth_id,
+            'userName': 'anon' + random_oauth_id
+        })
+        return self._download_json(
+            'https://ww.api.iheart.com/api/v1/account/loginOrCreateOauthUser',
+            current_id, "Registering temporary user", data=register_user_values,
+            headers={'Accept': 'application/json, text/plain, */*',
+                     'X-hostName': 'webapp.WW'})
+
+    def _get_streams_info(self, podcast_id, episode_ids, temp_user, current_id):
+        streams_values = json.dumps({
+            'contentIds': episode_ids,
+            'hostName': 'webapp.WW',
+            'playedFrom': 6,  # Not sure about the meaning behind this, Firefox uses 6
+            'stationId': compat_str(podcast_id),
+            'stationType': 'PODCAST'
+        }).encode('utf-8')
+        return self._download_json(
+            'https://ww.api.iheart.com/api/v2/playback/streams',
+            current_id, "Requesting stream info", data=streams_values,
+            headers={'Content-Type': 'application/json;charset=utf-8',
+                     'X-Session-Id': temp_user['sessionId'],
+                     'X-User-Id': temp_user['profileId']})
+
+    def _get_title(self, webpage):
+        podcast_title = self._html_search_meta(
+            ['og:title', 'title', 'twitter:title'],
+            webpage, 'title', default=None)
+        return re.sub(r' \| iHeartRadio$', '', podcast_title)
+
+    def _extract_episode(self, item_info, display_id, podcast_id, podcast_title):
+        content_info = item_info['content']
+
+        thumbnails = [{
+            'url': 'https://i.iheart.com/v3/catalog/podcast/%s' % podcast_id,
+            'width': 3000,
+            'height': 3000
+        }]
+
+        # They have an API that dynamically generates images of a needed size
+        # This is a list of "standard" sizes which are used by the web app
+        for size in [75, 240, 480]:
+            thumbnails.append({
+                'url': 'https://i.iheart.com/v3/catalog/podcast/%(id)s?ops=fit(%(size)d, %(size)d)' % {
+                    'id': podcast_id, 'size': size},
+                'width': size,
+                'height': size
+            })
+
+        # Release date timestamp is in milliseconds
+        release_date = content_info.get('startDate')
+        if isinstance(release_date, Number) and release_date > 2000000000:
+            release_date /= 1000
+
+        # Remove analytics from stream URL (optional)
+        streamUrl = item_info['streamUrl']
+        streamUrl = re.sub(r'(?:www\.)?podtrac\.com/pts/redirect\.[\w]*/', '', streamUrl)
+        streamUrl = re.sub(r'chtbl\.com/track/[\w]*/', '', streamUrl)
+        streamUrl = re.sub(r'\?source=[\w]*', '', streamUrl)
```

Comment:
this should done separatly from this PR(it's used in multiple services that are supported by youtube-dl, so it should be done in generic way).

Reply:
I couldn't find any other mentions of "podtrac" or "chbtl" in other extractors. But yeah, I guess I shouldn't be doing that in this extractor.